scarb/tests/lint.rs

```diff
@@ -518,3 +518,65 @@ fn test_missing_feature() {
             note: to use features, you need to define [features] section in Scarb.toml
         "#});
 }
+
+#[test]
+fn lint_selected_targets() {
```

Comment:
I don't understand why `scarb lint --workspace` does not lint all packages in the workspace anyway? 🤔 This sounds like what should happen without the need to explicitly list target names.

Reply:
Tbh it sounds reasonable to me, that passing `--workspace` alongside with `--target-names` will lint only the targets (including the whole workspace), that were specified in the `--target-names`. Also, I'm not sure, but as I looked roughly through the `build` logic, it acts the same. It first filters out packages based on `pakcage_filter` arg (which also includes `--workspace` inside) https://github.com/software-mansion/scarb/blob/5dc5cceaa26d103fea70f39153ce60ba5f7358e5/scarb/src/bin/scarb/commands/build.rs#L11, and later filters out the targets here: https://github.com/software-mansion/scarb/blob/5dc5cceaa26d103fea70f39153ce60ba5f7358e5/scarb/src/ops/compile.rs#L147.

What do you think?

Reply:
Oh, I think I was confused. It does indeed work correctly. Sorry 🙇